experimental/ComplexReflectionGroups/src/complex_reflection_group.jl

```
@@ -0,0 +1,182 @@
+        else
+            error("Specified model not found")
+        end
+
+        # set attributes that are already known from type
+        set_attribute!(matgrp, :order, order(C))
+        set_attribute!(matgrp, :is_complex_reflection_group, true)
+        set_attribute!(matgrp, :complex_reflection_group_type, C)
+        set_attribute!(matgrp, :complex_reflection_group_model, [Cmodel])
+        set_attribute!(matgrp, :is_irreducible, true)
+
+        # add to list
+        push!(component_groups, matgrp)
+        push!(modellist, model)
+    end
+
+    if length(component_groups) == 1
+        # treat this as a special case because direct_product([G]) will have type direct
+        # product which looks weird for a single group.
+        return component_groups[1]
+    else
+        matgrp = direct_product(component_groups)
+
+        set_attribute!(matgrp, :order, order(G))
+        set_attribute!(matgrp, :is_complex_reflection_group, true)
+        set_attribute!(matgrp, :complex_reflection_group_type, G)
+        set_attribute!(matgrp, :complex_reflection_group_model, modellist)
+        set_attribute!(matgrp, :is_irreducible, false)
+
+        return matgrp
+    end
+
+end
+
+# Convenience constructors
+complex_reflection_group(i::Int, model::Symbol=:default) = complex_reflection_group(ComplexReflectionGroupType(i), model)
+
+complex_reflection_group(m::Int, p::Int, n::Int, model::Symbol=:default) = complex_reflection_group(ComplexReflectionGroupType(m,p,n), model)
+
+complex_reflection_group(X::Vector, model::Symbol=:default) = complex_reflection_group(ComplexReflectionGroupType(X), model)
+
+
+
+
+function complex_reflection_group_type(G::MatrixGroup)
+    if has_attribute(G, :complex_reflection_group_type)
+        return get_attribute(G, :complex_reflection_group_type)
+    end
+    return nothing
+    # this should be upgraded later to work with a general matrix group (identifying the
+    # type from scratch is not so easy though)
```

Maybe throw an error until then instead of returning `nothing`.